internal/handlers/common/filter.go

```diff
@@ -157,14 +157,24 @@ func filterFieldExpr(fieldValue any, expr *types.Document) (bool, error) {
 			continue
 		}
 
-		exprValue := must.NotFail(expr.Get(exprKey))
+		exprValue := must.NotFail(expr.Get(exprKey)).(any)
 
 		switch exprKey {
 		case "$eq":
 			// {field: {$eq: exprValue}}
-			// TODO regex
-			if compareScalars(fieldValue, exprValue) != equal {
-				return false, nil
+			_, isValueDocument := fieldValue.(*types.Document)
+			_, isValueArray := fieldValue.(*types.Array)
+			exprValueRegex, isExprValueRegex := exprValue.(types.Regex)
+			fieldValueRegex, isFieldValueRegex := fieldValue.(types.Regex)
+			switch {
+			case isValueDocument, isValueArray:
+				return compare(fieldValue, exprValue) == equal, nil
+
+			case isExprValueRegex && isFieldValueRegex:
+				return filterFieldRegex(fieldValueRegex, exprValueRegex)
+
+			default:
+				return compareScalars(fieldValue, exprValue) == equal, nil
```

Comment:
in case if one of variables is regex, it will panic what you probably don't want. Isn't it?

Reply:
logic has been rewritten, panic should not appear